### DISCUSSION TITLE
If openjdk and adoptopenjdk8 are deprecated in favor or temurin how get jdk8?

### DISCUSSION THREAD

**Discussion by jkleckner** [Homebrew/discussions#2172] [Casks] [answered]
The title sort of says it.  This discussion [1] indicates that both `openjdk` and `adoptopenjdk8` are deprecated in favor of `temurin`.

Trying `brew install temurin@8` doesn't work - is there a recommended way to get the current jdk8 build?

The adoptium project does provide a jdk8 current build [2].

Cc: @SMillerDev 

[1] https://github.com/Homebrew/discussions/discussions/2065
[2] https://adoptopenjdk.net/?variant=openjdk8&jvmVariant=hotspot

**Comment by SMillerDev** [comment]
Openjdk isn't deprecated, and I'm not really sure how the discussion would suggest it to be.

**Comment by jkleckner** [comment]
I was basing that on the @chriskilding comment in #2065 :  `Recently the OpenJDK formula was deprecated; development was stopped on August 1st. They recommend changing to the Temurin brew casks instead.`

In addition, I installed the version with `brew install openjdk@8` and noted that it was 8 point releases behind which I took to mean it was lagging adoptium.  It gives `build 25.302-b00` with a date of August 14 whereas the adoptium indicates `jdk8u302-b08`.

**Reply by Bo98** [reply]
The OpenJDK cask in AdoptOpenJDK's Homebrew tap is deprecated, but not the OpenJDK formula shipped with homebrew-core which has nothing to do with AdoptOpenJDK.

Homebrew's `openjdk@8` isn't actually outdated. The `b` part of the version defaults to 0 unless it is set manually (it's not something contained in the source code). Homebrew's build doesn't set it. There's an argument that we maybe should for correctness, though note that we only ever ship GA releases so it'll never be any other value but the latest (previous `b` versions are betas).

**Comment by cho-m** [comment, marked as answer]
The answer in that discussion explains OpenJDK **Formulae** are unrelated to AdoptOpenJDK/Adoptium/etc **Casks**.

The previous discussion was related to how "AdoptOpenJDK" organization was replaced by "Eclipse Adoptium" (Temurin), which provides pre-built JDK distributions that Homebrew packages as **Casks**.

On the other hand, the OpenJDK **Formulae** are installed from source (with pre-built bottles by Homebrew) with no relation to AdoptOpenJDK/Adoptium/etc.

---

The following are all available Java distributions with a fresh Homebrew install (no extra taps):
- Formula [`openjdk@8`](https://formulae.brew.sh/formula/openjdk@8) - JDK 8 distribution with bottles built by **Homebrew**
- Formula [`openjdk@11`](https://formulae.brew.sh/formula/openjdk@11) - JDK 11 distribution with bottles built by **Homebrew**
- Formula [`openjdk`](https://formulae.brew.sh/formula/openjdk) - latest JDK (currently 17) distribution with bottles built by **Homebrew**
- Cask [`oracle-jdk`](https://formulae.brew.sh/cask/oracle-jdk) - Pre-built JDK distribution by **Oracle** that tracks their latest release (currently 17)
- Cask [`microsoft-openjdk`](https://formulae.brew.sh/cask/microsoft-openjdk) - Pre-built JDK distribution by **Microsoft** that tracks their latest release (currently 16)
- Cask [`temurin`](https://formulae.brew.sh/cask/temurin) - Pre-built JDK distribution by **Eclipse Adoptium** (successor to **AdoptOpenJDK**) that tracks their latest release (currently 16)
- Cask [`zulu`](https://formulae.brew.sh/cask/zulu) - Pre-built JDK distribution by **Azul** that tracks their latest release (currently 17)

The following are versioned Java Casks available after running `brew tap homebrew/cask-versions`
- Cask [`temurin8`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/temurin8.rb) - Pre-built JDK 8 distribution by **Eclipse Adoptium**
- Cask [`temurin11`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/temurin11.rb) - Pre-built JDK 11 distribution by **Eclipse Adoptium**
- Cask [`microsoft-openjdk11`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/microsoft-openjdk11.rb) - Pre-built JDK 11 distribution by **Microsoft**
- Cask [`zulu7`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/zulu7.rb) - Pre-built JDK 7 distribution by **Azul**
- Cask [`zulu8`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/zulu8.rb) - Pre-built JDK 8 distribution by **Azul**
- Cask [`zulu11`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/zulu11.rb) - Pre-built JDK 11 distribution by **Azul**
- Cask [`zulu13`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/zulu13.rb) - Pre-built JDK 13 distribution by **Azul**
- Cask [`zulu15`](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/zulu15.rb) - Pre-built JDK 15 distribution by **Azul**

**Reply by jkleckner** [reply]
Thank you all for definitively answering the question and especially @cho-m  for detailing all the variations!
